Content.Server/Destructible/Thresholds/Behaviors/SpillBehavior.cs

```diff
@@ -1,42 +1,60 @@
-        {
-            var solutionContainerSystem = system.EntityManager.System<SharedSolutionContainerSystem>();
-            var spillableSystem = system.EntityManager.System<PuddleSystem>();
+    /// <summary>
+    /// When triggered, spills the entity's solution onto the ground.
+    /// Will first try to use the solution from a SpillableComponent if present,
+    /// otherwise falls back to the solution specified in the behavior's data fields.
+    /// The solution is properly drained/split before spilling to prevent double-spilling with other behaviors.
+    /// </summary>
+    /// <param name="owner">Entity whose solution will be spilled</param>
+    /// <param name="system">System calling this behavior</param>
+    /// <param name="cause">Optional entity that caused this behavior to trigger</param>
+    public void Execute(EntityUid owner, DestructibleSystem system, EntityUid? cause = null)
+    {
+        var solutionContainerSystem = system.EntityManager.System<SharedSolutionContainerSystem>();
+        var spillableSystem = system.EntityManager.System<PuddleSystem>();
+        var coordinates = system.EntityManager.GetComponent<TransformComponent>(owner).Coordinates;
 
-            var coordinates = system.EntityManager.GetComponent<TransformComponent>(owner).Coordinates;
+        Solution targetSolution;
 
-            if (system.EntityManager.TryGetComponent(owner, out SpillableComponent? spillableComponent) &&
-                solutionContainerSystem.TryGetSolution(owner, spillableComponent.SolutionName, out _, out var compSolution))
-            {
-                spillableSystem.TrySplashSpillAt(owner, coordinates, compSolution, out _, false, user: cause);
-            }
-            else if (Solution != null &&
-                     solutionContainerSystem.TryGetSolution(owner, Solution, out _, out var behaviorSolution))
-            {
-                spillableSystem.TrySplashSpillAt(owner, coordinates, behaviorSolution, out _, user: cause);
-            }
+        // First try to get solution from SpillableComponent
+        if (system.EntityManager.TryGetComponent(owner, out SpillableComponent? spillableComponent) &&
+            solutionContainerSystem.TryGetSolution(owner, spillableComponent.SolutionName, out var solution, out var compSolution))
+        {
+            // If entity is drainable, drain the solution. Otherwise just split it.
+            // Both methods ensure the solution is properly removed.
+            targetSolution = system.EntityManager.HasComponent<DrainableSolutionComponent>(owner)
+                ? solutionContainerSystem.Drain((owner, system.EntityManager.GetComponent<DrainableSolutionComponent>(owner)), solution.Value, compSolution.Volume)
+                : compSolution.SplitSolution(compSolution.Volume);
         }
+        // Fallback to solution specified in behavior data
+        else if (Solution != null &&
```

Comment:
This `else if` block should be able to be deduplicated by chaining the `TryGetSolution` calls in the `if` statement.